internal/datacoord/clustering_compaction_manager.go

```diff
@@ -224,252 +206,310 @@ func (t *ClusteringCompactionManager) gc() error {
-					TaskID: analyzeTaskID,
-					State:  indexpb.JobState_JobStateInit,
-				},
-			})
-			log.Info("submit analyze task", zap.Int64("id", analyzeTaskID))
-
-			var analyzeTask *indexpb.AnalyzeTask
-			analyzeFinished := func() bool {
-				analyzeTask = t.meta.analyzeMeta.GetTask(analyzeTaskID)
-				log.Debug("check analyze task state", zap.Int64("id", analyzeTaskID), zap.String("state", analyzeTask.State.String()))
-				if analyzeTask.State == indexpb.JobState_JobStateFinished ||
-					analyzeTask.State == indexpb.JobState_JobStateFailed {
-					return true
-				}
-				return false
-			}
-			for !analyzeFinished() {
-				// respect context deadline/cancel
-				select {
-				case <-t.ctx.Done():
-					return nil
-				default:
-				}
-				time.Sleep(1 * time.Second)
-			}
-			log.Info("get analyzeTask", zap.Any("analyzeTask", analyzeTask))
-			if analyzeTask.State == indexpb.JobState_JobStateFinished {
-				//version := int64(0) // analyzeTask.Version
-				//plan.AnalyzeResultPath = path.Join(metautil.JoinIDPath(analyzeTask.TaskID, analyzeTask.Version))
-				if analyzeTask.GetCentroidsFile() == "" && len(analyzeTask.GetOffsetMapping()) == 0 {
-					job.state = completed
-					return t.saveJob(job)
-				}
-				plan.CentroidFilePath = analyzeTask.GetCentroidsFile()
-				plan.OffsetMappingFiles = analyzeTask.GetOffsetMapping()
-				offSetSegmentIDs := make([]int64, 0)
-				for _, segID := range analyzeTask.SegmentIDs {
-					offSetSegmentIDs = append(offSetSegmentIDs, segID)
-				}
-				plan.AnalyzeSegmentIds = offSetSegmentIDs
-			}
-		}
+	task.State = datapb.CompactionTaskState_compact_pipelining
+	log.Info("send compaction task to execute", zap.Int64("triggerID", task.GetTriggerId()),
+		zap.Int64("planID", task.GetPlanId()),
+		zap.Int64("collectionID", task.GetCollectionId()),
+		zap.Int64("partitionID", task.GetPartitionId()),
+		zap.Int64s("inputSegments", task.InputSegments))
+	return nil
+}
 
-		//shouldDo, err := t.shouldDoClusteringCompaction(analyzeResult)
-		//if err != nil {
-		//	log.Warn("failed to decide whether to execute this compaction plan", zap.Int64("planID", plan.PlanID), zap.Int64s("segmentIDs", segIDs), zap.Error(err))
-		//	continue
-		//}
-		//if !shouldDo {
-		//	log.Info("skip execute compaction plan", zap.Int64("planID", plan.PlanID))
-		//	continue
-		//}
-
-		trigger := &compactionSignal{
-			id:           job.triggerID,
-			collectionID: job.collectionID,
-			partitionID:  plan.SegmentBinlogs[0].PartitionID,
-		}
-		err = t.compactionHandler.execCompactionPlan(trigger, plan)
-		if err != nil {
-			log.Warn("failed to execute compaction plan", zap.Int64("planID", plan.GetPlanID()), zap.Int64s("segmentIDs", segIDs), zap.Error(err))
+func (t *ClusteringCompactionManager) summaryCompactionTaskState(compactionTasks []*datapb.CompactionTask) (state commonpb.CompactionState, executingCnt, pipeliningCnt, completedCnt, failedCnt, timeoutCnt, initCnt, analyzingCnt, analyzedCnt, indexingCnt, indexedCnt int) {
```

Comment:
Use a struct to wrap the returned result.
and from the services.go:1126, I think it can only return `executingCnt`, `completedCnt`, `timeoutCnt`, `failedCnt` and `state`. and log all values in this function.